libfwupdplugin/fu-sbatlevel-section-firmware.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2023 Canonical Ltd.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FuFirmware"
+
+#include "config.h"
+
+#include "fu-bytes.h"
+#include "fu-csv-firmware.h"
+#include "fu-mem.h"
+#include "fu-sbatlevel-section-firmware.h"
+#include "fu-sbatlevel-section-struct.h"
+
+G_DEFINE_TYPE(FuSbatlevelSectionFirmware, fu_sbatlevel_section_firmware, FU_TYPE_FIRMWARE);
+
+static gboolean
+fu_sbatlevel_section_add_entry(FuFirmware *firmware,
+			       GBytes *fw,
+			       const guint8 *buf,
+			       gsize bufsz,
```

Comment:
you don't need these if you have `fw` -- just do `const guint8 *buf = g_bytes_get_data(fw, &bufsz);` again